Src/ILGPU/Runtime/Cuda/CudaIpcMemoryBuffer.cs

```diff
@@ -0,0 +1,173 @@
+        {
+            if (targetView.GetAcceleratorType() != AcceleratorType.Cuda)
+            {
+                throw new NotSupportedException(
+                    RuntimeErrorMessages.NotSupportedTargetAccelerator);
+            }
+
+            using var binding = stream?.Accelerator.BindScoped();
+            CudaException.ThrowIfFailed(
+                CurrentAPI.Memset(
+                    targetView.LoadEffectiveAddressAsPtr(),
+                    value,
+                    new IntPtr(targetView.LengthInBytes),
+                    stream));
+        }
+
+        /// <summary>
+        /// Performs a Cuda copy operation.
+        /// </summary>
+        /// <typeparam name="T">The element type.</typeparam>
+        /// <param name="stream">
+        /// The Cuda stream to use (can be null to preserve backwards compatibility).
+        /// </param>
+        /// <param name="sourceView">The source view to copy from.</param>
+        /// <param name="targetView">The target view to copy to.</param>
+        public static void CudaCopy<T>(
+            CudaStream? stream,
+            in ArrayView<T> sourceView,
+            in ArrayView<T> targetView)
+            where T : unmanaged
+        {
+            var sourceType = sourceView.GetAcceleratorType();
+            var targetType = targetView.GetAcceleratorType();
+
+            if (sourceType == AcceleratorType.OpenCL ||
+                targetType == AcceleratorType.OpenCL)
+            {
+                throw new NotSupportedException(
+                    RuntimeErrorMessages.NotSupportedTargetAccelerator);
+            }
+
+            var sourceAddress = sourceView.LoadEffectiveAddressAsPtr();
+            var targetAddress = targetView.LoadEffectiveAddressAsPtr();
+            var length = new IntPtr(targetView.LengthInBytes);
+
+            // a) Copy from CPU to GPU
+            // b) Copy from GPU to CPU
+            // c) Copy from GPU to GPU
+            using var binding = stream?.Accelerator.BindScoped();
+            CudaException.ThrowIfFailed(
+                CurrentAPI.MemcpyAsync(
+                    targetAddress,
+                    sourceAddress,
+                    length,
+                    stream));
+        }
+
+        #endregion
+
+        #region Instance
+
+        /// <summary>
+        /// Constructs a new Cuda buffer.
+        /// </summary>
+        /// <param name="accelerator">The accelerator.</param>
+        /// <param name="ipcMemHandle">A IPC memory handle from another process.</param>
+        /// <param name="length">The length of this buffer.</param>
+        /// <param name="elementSize">The element size.</param>
+        /// <param name="flags">The flags to use.</param>
+        public CudaIpcMemoryBuffer(
+            CudaAccelerator accelerator,
+            CudaIpcMemHandle ipcMemHandle,
+            long length,
+            int elementSize,
+            CudaIpcMemFlags flags = CudaIpcMemFlags.None)
```

Comment:
I would suggest making the `flags` mandatory, so that the caller has to provide it. It will primarily be used internally anyway.

Reply:
Agreed, I made it mandatory. What I'm not sure about is, whether we should even add them to the "convenience" methods. The currently only flag `LazyEnablePeerAccess` will enable peer access if the underlying memory is allocated on a different device than the device used by the current context.
Obviously I'm quite new to this library, but my gut says this would be a rather niche case not worth complicating the API for.

Reply:
Well, I spoke too soon. Without `LazyEnablePeerAccess` IPC won't work for me even with the same device. I strongly suspect it relates to me having an SLI setup with two GPUs, but I was unable to find a definite answer.  
Unsurprisingly, I'm now strongly in favor of exposing the flags. I also added my finding in the remarks.